nixos/modules/system/boot/stage-2.nix

```diff
@@ -51,14 +61,25 @@ in
         '';
       };
 
-      readOnlyNixStore = mkOption {
-        type = types.bool;
-        default = true;
+      nixStoreMountOpts = mkOption {
+        type = types.listOf types.nonEmptyStr;
+        default = [
+          "ro"
+          "nodev"
+          "nosuid"
+        ];
+        apply = opts: lib.concatStringsSep " " (map lib.escapeShellArg opts);
         description = ''
-          If set, NixOS will enforce the immutability of the Nix store
-          by making {file}`/nix/store` a read-only bind
-          mount.  Nix will automatically make the store writable when
-          needed.
+          Defines the mount options used on a bind mount for the {file}`/nix/store`.
+          This affects the whole system except the nix store daemon, which will undo the bind mount.
+
+          `ro` enforces immutability of the Nix store.
+
+          `nosuid` enforces no suid binaries of the Nix store get executed.
+          The Nix store should already not contain any suid binaries at any point.
+
+          `nodev` enforces no device mappers in the Nix store are valid.
+          The Nix store should already not contain device mappers at any point.
```

Comment:
I don't think we need to document what VFS options do.

Reply:
I intended to document why the defaults are there and why the defaults shouldn't break anything. Especially with `readOnlyNixStore` option removed, it might make sense to be a bit more verbose about it. But it didn't need to be *that* verbose, i shortened it significantly.